### ISSUE TITLE
Get the registration ID from OGN Device Data Base

### ISSUE THREAD

**Issue by acasadoalonso** [PepperJo/stratux#1]
We should get the registration from the OGN Device Data Base (DDB) instead of displaying the FLARM ID we should display the registration D-xxxx or EC-ACF
The process could be to download the DDB from ddb.glidernet.org on a periodic base and replace the Flarm ID with the registration ID when we displayed on the WEB page.

**Comment by PepperJo** [comment]
Good idea. @cyoung is working on a FLARM build for the new v3 radio, so I hope we can abandon this fork soon and move to upstream stratux. As soon as this happened I'm happy to look into this feature.

**Comment by acasadoalonso** [comment]
Jonas,

Thanks for the quick reply.

As a hint you can download the OGN DDB on JSON format this way:

wget -O ognddbdata.json ddb.glidernet.org/download/?j=1 <http://ddb.glidernet.org/download/?j=1> 

and the output is similar to this:

{"devices":[{"device_type":"F","device_id":"000000","aircraft_model":"HPH 304CZ-17","registration":"OK-7777","cn":"KN","tracked":"Y","identified":"Y"},{"device_type":"O"
,"device_id":"000001","aircraft_model":"Paraglider","registration":"","cn":"","tracked":"Y","identified":"Y"},{"device_type":"F","device_id":"000002","aircraft_model":"L
S-6 18","registration":"OY-XRG","cn":"G2","tracked":"Y","identified":"Y"},{"device_type":"F","device_id":"00000D","aircraft_model":"Ka-8","registration":"D-1749","cn":"W
5","tracked":"Y","identified":"Y"},{"device_type":"O","device_id":"000015","aircraft_model":"Paraglider","registration":"36445","cn":"","tracked":"Y","identified":"Y"},{
"device_type":"F","device_id":"0000FD","aircraft_model":"Taurus","registration":"F-JRDN","cn":"DN","tracked":"Y","identified":"Y"},{"device_type":"F","device_id":"000114
","aircraft_model":"","registration":"","cn":"","tracked":"N","identified":"N"},{"device_type":"F","device_id":"000146","aircraft_model":"Paraglider","registration":"ant
wilk","cn":"","tracked":"Y","identified":"Y"},{"device_type":"F","device_id":"000287","aircraft_model":"Paraglider","registration":"32712","cn":"","tracked":"Y","identif
ied":"Y"},{"device_type":"F","device_id":"000296","aircraft_model":"Unknown","registration":"D-IEST","cn":"","tracked":"Y","identified":"Y"},{"device_type":"F","device_i
d":"000333","aircraft_model":"ASG-29","registration":"N229CW","cn":"S2H","tracked":"Y","identified":"Y"},{"device_type":"F","device_id":"000338","aircraft_model":"Unknow
n","registration":"","cn":"","tracked":"Y","identified":"Y"},{"device_type":"F","device_id":"000368","aircraft_model":"PIK-20 B","registration":"C-GORE","cn":"GP","track
ed":"Y","identified":"Y"},{"device_type":"F","device_id":"000451","aircraft_model":"ASG-29","registration":"N232FG","cn":"FG","tracked":"Y","identified":"Y"},{"device_ty
pe":"F","device_id":"000457","aircraft_model":"ASW-20","registration":"OK-9496","cn":"96","tracked":"Y","identified":"Y"},{"device_type":"F","device_id":"000503","aircra
ft_model":"SZD-48-1 Jantar Std 2","registration":"SP-3152","cn":"DG","tracked":"Y","identified":"Y"},{"device_type":"F","device_id":"000597","aircraft_model":"LS-6","reg
istration":"D-8997","cn":"IV","tracked":"Y","identified":"Y”},


And around line 380 of stratux/main/flarm.go 
you can substitute the FlarmID by the registration ID using the data from the JSON structure.

Tell me if you need further details, I am maintaining for the time being the code for the OGN Device database app: ddb.glidernet.org <http://ddb.glidernet.org/> 
Cheers from rainy Madrid today.

AC/.

**Comment by PepperJo** [comment]
Thanks for the info. I'm also happy to merge pull request if you want to give it a go. I have another pending item that I want to look at first (per client switch between alt MSL and height above WGS-84 cf. https://github.com/cyoung/stratux/pull/749)

**Comment by acasadoalonso** [comment]
I never programed in GO, however perhaps this is a good occasion to try it … I will let you know !!!!
